### PR TITLE
[TACHYON-960] Make some parameters in master module configurable

### PR DESCRIPTION
**Pull request by uronce-cc**
https://tachyon.atlassian.net/browse/TACHYON-960

### CODE REVIEW

File: clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java [hunk 1]
```diff
@@ -382,7 +382,6 @@ public FileStatus getFileStatus(Path path) throws IOException {
    * @see org.apache.hadoop.fs.FileSystem#createFileSystem(java.net.URI,
    *      org.apache.hadoop.conf.Configuration)
    */
-  @Override
```
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java]: This seems unrelated to the rest of the PR?

File: common/src/main/resources/tachyon-default.properties [hunk 2]
```diff
@@ -42,6 +42,10 @@ tachyon.host.resolution.timeout.ms=5000
 
 # Master properties
 tachyon.master.journal.folder=${tachyon.home}/journal/
+tachyon.master.journal.formatter.class=tachyon.master.journal.JsonJournalFormatter
+tachyon.master.journal.tailer.sleep.time.ms=1
```
Comment (gpang) [inline common/src/main/resources/tachyon-default.properties]: this should be 1000 or greater, since this should be in ms.

File: common/src/main/resources/tachyon-default.properties [hunk 3]
```diff
@@ -42,6 +42,10 @@ tachyon.host.resolution.timeout.ms=5000
...
+tachyon.master.journal.tailer.shutdown.quiet.wait.time.ms=5
```
Comment (gpang) [inline common/src/main/resources/tachyon-default.properties]: This should be 5000 or greater since this is in ms.

File: common/src/main/resources/tachyon-default.properties [hunk 4]
```diff
@@ -42,6 +42,10 @@ tachyon.host.resolution.timeout.ms=5000
...
+tachyon.master.journal.max.log.size.bytes=10MB
```
Comment (gpang) [inline common/src/main/resources/tachyon-default.properties]: All 4 of these new config parameters should be documented in the docs/.

File: integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java [hunk 5]
```diff
@@ -158,13 +157,26 @@ public final void before() throws Exception {
   }
 
   /**
+   * Assume {@link #before} has been called, then we need to first stop the cluster and start the
+   * cluster again after reconfiguring {@link Constants#MASTER_JOURNAL_MAX_LOG_SIZE_BYTES}.
+   *
+   * @param bytes max log size in bytes
+   */
+  private void configureMaxLogSizeAndRestart(int bytes) throws Exception {
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java]: Can't you achieve the same thing by passing in a TachyonConf into start() of LocalTachyonCluster?

**Comment by gpang** [comment]
Thanks for this improvement! I left some comments.

File: clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java (hunk 1 above)
Comment (uronce-cc) [inline clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java]: Yes, but it blocks compilation.

File: integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java (hunk 5 above)
Comment (uronce-cc) [inline integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java]: But in `before()`, a cluster has already been started, so I need to stop it first and then restart with a new `TachyonConf`.

**Comment by uronce-cc** [comment]
@gpang Updated, PTAL.

**Comment by jsimsa** [comment]
LGTM

File: integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java (hunk 5 above)
Comment (gpang) [inline integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java]: Can't you just create a conf, and pass it into start(), before the cluster starts?

File: docs/Configuration-Settings.md [hunk 6]
```diff
@@ -168,6 +168,23 @@ number.
...
 </tr>
 <tr>
+  <td>tachyon.master.journal.formatter.class</td>
+  <td>tachyon.master.journal.JsonJournalFormatter</td>
+  <td>The class to serialize the journal in a specified format.</td>
+</tr>
+<tr>
+  <td>tachyon.master.journal.tailer.sleep.time.ms</td>
+  <td>1000</td>
+  <td>Time(in miniseconds) to sleep for standby master when it cannot find anything new from leader
```
Comment (gpang) [inline docs/Configuration-Settings.md]: milliseconds

File: docs/Configuration-Settings.md [hunk 7]
```diff
@@ -168,6 +168,23 @@ number.
...
+    master's journal.</td>
+</tr>
+<tr>
+  <td>tachyon.master.journal.tailer.shutdown.quiet.wait.time.ms</td>
+  <td>5000</td>
+  <td>Before the standby master shuts down its tailer thread, there should be no update to the
+    leader master's journal in this specified time period(in miniseconds).</td>
```
Comment (gpang) [inline docs/Configuration-Settings.md]: milliseconds

**Comment by gpang** [comment]
I left a few more comments. Thanks!

File: integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java (hunk 5 above)
Comment (uronce-cc) [inline integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java]: The cluster starts in `before()`, so I have to first stop it.  Could you provide some suggestions on how to accomplish what you want?

File: docs/Configuration-Settings.md (hunk 7 above)
Comment (uronce-cc) [inline docs/Configuration-Settings.md]: Updated.

File: integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java (hunk 5 above)
Comment (gpang) [inline integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java]: In before(), the cluster is started with mLocalTachyonCluster.start(). I think what you can do is create a TachyonConf conf, and set the variable like you do here, and then call mLocalTachyonCluster.start(conf); in before().

Although, then it means all the tests need the same log size. Is that acceptable for these tests?

Comment (uronce-cc) [inline integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java]: Yeah, currently, it is acceptable. I've changed according to this idea.

**Comment by gpang** [comment]
LGTM
